contracts/src/TeleporterTokenDestination.sol

```diff
@@ -445,6 +446,10 @@ abstract contract TeleporterTokenDestination is
                 abi.decode(bridgeMessage.payload, (SingleHopSendMessage));
             _withdraw(payload.recipient, payload.amount);
         } else if (bridgeMessage.messageType == BridgeMessageType.SINGLE_HOP_CALL) {
+            // Decode the call message payload.
+            // Since the message may be part of a multi-hop message, we cannot
+            // authenticate the sender information in the payload against the Teleporter
+            // arguments since they may originate from different blockchains.
```

Comment:
```suggestion
            // The {sourceBlockchainID}, and {originSenderAddress} specified in the message
            // payload will not match the sender of this Teleporter message in the case of a 
            // multi-hop message. Since Teleporter message are only received from the specified
            // source contract, no additional authentication is needed on the payload values.
```

Nit: Saying "we can't authenticate the sender information" makes it seem like it may not be trustworthy or represent a vulnerability. I think this wording makes it more clear that no such checks are needed here.